src/embed.ts

```diff
@@ -18,31 +18,40 @@ import type { DocParser } from 'types/parserContext';
 
 // ----- Types ----- //
 
-const enum EmbedKind {
-	Generic,
-	Instagram,
-	Spotify,
-	YouTube,
+enum EmbedKind {
+	Generic = 'Generic',
+	Instagram = 'Instagram',
+	Spotify = 'Spotify',
+	YouTube = 'YouTube',
 }
 
 interface YouTube {
 	kind: EmbedKind.YouTube;
 	id: string;
 	width: number;
 	height: number;
+	source: Option<string>;
+	sourceDomain: Option<string>;
+	tracking: EmbedTracksType;
 }
 
 interface Spotify {
 	kind: EmbedKind.Spotify;
 	src: string;
 	width: number;
 	height: number;
+	source: Option<string>;
+	sourceDomain: Option<string>;
+	tracking: EmbedTracksType;
 }
 
 interface Instagram {
 	kind: EmbedKind.Instagram;
 	id: string;
 	caption: Option<string>;
+	source: Option<string>;
+	sourceDomain: Option<string>;
+	tracking: EmbedTracksType;
```

Comment:
If we _know_ these are from YouTube, Spotify and Instagram don't we already know the `source` and `sourceDomain`?

Reply:
Yes its a good point, I guess there is a certain appeal to all of the embeds working in the same way.

Let me take a look and make sure this wouldnt complicate the code.

Reply:
Ive made this change, it adds a bit of complication to the embedWrapper component but I think it an improvement overall.